### PR TITLE
PodAffinity default and reconcile changes #862 #863

### LINKED ISSUE #862: Default Affinity settings are not reported into Infinispan CR

### LINKED ISSUE #863: NodeAffinity changes are not reconcilied to the statefulset

### PR DESCRIPTION
**Pull request by rigazilla**
Fixes #862
Fixes #863

### CODE REVIEW

**Review by dmvolod — APPROVED** [review]

File: pkg/controller/infinispan/infinispan_controller.go [hunk 1]
```diff
@@ -159,6 +159,7 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 	err := r.update(infinispan, func() {
 		// Apply defaults and endpoint encryption settings if not already set
 		infinispan.ApplyDefaults()
+		infinispan.Spec.Affinity = podAffinity(infinispan, PodLabels(infinispan.Name))
```
Comment (dmvolod) [inline pkg/controller/infinispan/infinispan_controller.go:162]: overall lgtm 
@rigazilla do we really need this defaults for any user who need this or not?

Comment (rigazilla) [reply pkg/controller/infinispan/infinispan_controller.go:162]: @dmvolod yes, we're following this defaulting guidelines doc:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#defaulting

Comment (dmvolod) [reply pkg/controller/infinispan/infinispan_controller.go:162]: Yes, sure.
Is it correct, that default Pod Affinity is set to `podAntiAffinity`
```
spec:
  affinity:
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
        - podAffinityTerm:
            labelSelector:
              matchLabels:
                app: infinispan-pod
                clusterName: example-infinispan
                infinispan_cr: example-infinispan
            topologyKey: kubernetes.io/hostname
          weight: 100
  container:
    cpu: 500m
    memory: 512Mi
```

Comment (rigazilla) [reply pkg/controller/infinispan/infinispan_controller.go:162]: oh, yep. that should be fine, I've just reported the default setting we already have before to the .Spec

**Comment by rigazilla** [comment]
merged, thanks!

**Pull request merged**